release/internal/imagescanner/scanner_test.go

```diff
@@ -0,0 +1,332 @@
+// Copyright (c) 2025 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package imagescanner
+
+import (
+	"bytes"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	approvals "github.com/approvals/go-approval-tests"
+	"github.com/sirupsen/logrus"
+)
+
+func TestConfigValid(t *testing.T) {
+	tests := []struct {
+		name     string
+		config   Config
+		expected bool
```

Comment:
nit: we could rename this to `expectedSuccess` (or `successExpected`) to make the intent clearer